src/kOS/Utilities/Utils.cs

```diff
@@ -275,5 +275,60 @@ public static CelestialBody KOSExtensionGetParentBody(this CelestialBody body)
                 parent = null;
             return parent;
         }
+
+        /// <summary>
+        /// Determines if a given case insensitive name corresponds with a defined resource, and
+        /// outputs the case sensitive name for easy access from Squad's lists.
+        /// </summary>
+        /// <param name="insensitiveName">case insensitive name</param>
+        /// <param name="fixedName">output case sensitive name</param>
+        /// <returns>true if a matching resource definition is found</returns>
+        public static bool IsResource(string insensitiveName, out string fixedName)
+        {
+            // eliminate enumerators, use index based access
```

Comment:
Oh yeah!  I tried to convert away from an enumerator at first, but then found out it doesn't support indexes.  Then I expected them to finish the enumerator purge so I left the code ready.  It never got purged apparently.  I'll remove the comment.